app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/common/Presenter.kt

```diff
@@ -0,0 +1,17 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common
+
+import io.reactivex.disposables.Disposable
+
+abstract class Presenter {
+
+    var subscriptions: List<Disposable> = emptyList()
+
+    // Here view might not be loaded yet
+    open fun onCreate() {}
+
+    open fun onStart() {}
```

Comment:
According to this correction, I made change and I separate presenter method and Actiity lifecycle. Right now inheritence is not used to specify when presenter method should be called.